### PR TITLE
Framework: Move test runner to common area, server-side suite

### PR DESCRIPTION
**Pull request by blowery**
Part of #3822.

Move things around so we can use the test runner for both client and server code.

Make up a new suite for the server tests.

Add in a utility to enable network access inside a test scope. Allows the devdocs tests to pass.

### CODE REVIEW

**Comment by blowery** [comment]
One little hiccup here:
- [ ] Move i18n tests in. They have a fancy makefile target that builds output files. Perhaps we can do that another way?

File: server/Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+test-%: i18n/test/out
+	@GREP="$*" make test
+
+i18n/test/out: i18n/test/examples
+		@$(BABEL) i18n/test/examples --out-dir i18n/test/out > /dev/null
+
+clean:
+		@rm -rf i18n/test/out
+
+.PHONY: test test-%
```
Comment (aduth) [inline server/Makefile]: `i18n/test/out` and `clean` in `.PHONY` ? While you're there, fix the indenting above :wink:

Comment (blowery) [inline server/Makefile]: the first one is a real dir, so we can leave it. fixing clean and indent.

File: server/tests.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"config": {
+		"test": [ "parser" ]
+	},
+	"devdocs": {
+		"test": [ "index" ]
+	},
+	"i18n": {
+		"test": [ "test" ]
+	},
+	"i18nlint": {
```
Comment (gziolo) [inline server/tests.json]: We are removing `i18nlint` folder with https://github.com/Automattic/wp-calypso/pull/3849.

File: server/Makefile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+NODE_PATH := $(BASE_DIR)/test:$(BASE_DIR)/server:$(BASE_DIR)/client
+BABEL := $(NODE_BIN)/babel
+
+test: i18n/test/out
+	@NODE_ENV=development NODE_PATH=$(NODE_PATH) $(BASE_DIR)/test/runner.js --reporter=$(REPORTER) $(addprefix --grep ,$(GREP))
+
+test-%: i18n/test/out
+	@GREP="$*" make test
+
+i18n/test/out: i18n/test/examples
```
Comment (gziolo) [inline server/Makefile]: Shouldn't we clean dir after each test run? We had that before: https://github.com/Automattic/wp-calypso/blob/d3fa3833a7aa9f27e7c67254c8276d11c6963947/server/i18n/Makefile.

**Comment by gziolo** [comment]
I rebased to resolve conflicts with master branch. I left one minor comment and prompt to sync with @aduth task that deprecates `i18nlint`. I'm giving :+1:

File: server/Makefile (hunk 3 above)
Comment (blowery) [inline server/Makefile]: That would clean it _before_ it ran. I'm not sure why we were cleaning after the run in the first place. Seems unnecessary?

Comment (blowery) [inline server/Makefile]: Either way, added it back in https://github.com/Automattic/wp-calypso/commit/a7d8c2c0157fe72cade3fe70bb21784f1a480052

Comment (gziolo) [inline server/Makefile]: :+1:

**Pull request merged**